XCUITests/ThirdPartySearchTest.swift

```diff
@@ -5,172 +5,181 @@
-            XCTAssertFalse(app.scrollViews.otherElements.buttons["developer.mozilla.org search"].exists)
         }
-    }
-    // Test failing in 10.3 due to timing issue when doing the last assert, it is done later than expected and so the text is different and fails
-    /*func testCustomEngineFromCorrectTemplate() {
-        let app = XCUIApplication()
+        app.buttons["AddSearch"].tap()
+        app.alerts["Add Search Provider?"].buttons["OK"].tap()
+        XCTAssertFalse(app.buttons["AddSearch"].isEnabled)
+        dismissKeyboardAssistant(forApp: app)
         
-        app.buttons["Menu"].tap()
+        let tabTrayButton = self.tabTrayButton(forApp: app)
+        tabTrayButton.tap()
+        app.buttons["TabTrayController.addTabButton"].tap()
+        app.textFields["url"].tap()
+        app.typeText("window")
+        
+        // For timing issue, we need a wait statement
+        waitforExistence(app.scrollViews.otherElements.buttons["developer.mozilla.org search"])
+        XCTAssert(app.scrollViews.otherElements.buttons["developer.mozilla.org search"].exists)
+        app.typeText("\r")
+        
+        // Go to settings and set MDN as the default
+        tabTrayButton.tap(force: true)
+        app.buttons["TabTrayController.menuButton"].tap()
         app.collectionViews.cells["Settings"].tap()
         let tablesQuery = app.tables
         tablesQuery.cells["Search"].tap()
-        app.tables.cells["customEngineViewButton"].tap()
-        app.textViews["customEngineTitle"].tap()
-        app.typeText("Feeling Lucky")
-        app.textViews["customEngineUrl"].tap()
-        app.typeText("http://www.google.com/search?q=%s&btnI")
         
-        app.navigationBars.buttons["customEngineSaveButton"].tap()
-
-        waitforExistence(app.navigationBars["Search"])
-        XCTAssert(app.navigationBars["Search"].buttons["Settings"].exists)
-
+        app.navigationBars["Search"].buttons["Edit"].tap()
+        tablesQuery.buttons["Delete developer.mozilla.org"].tap()
+        tablesQuery.buttons["Delete"].tap()
+        
         app.navigationBars["Search"].buttons["Settings"].tap()
         app.navigationBars["Settings"].buttons["AppSettingsTableViewController.navigationItem.leftBarButtonItem"].tap()
         
-        // Perform a search using a custom search engine
-        tabTrayButton(forApp: app).tap()
-        app.buttons["TabTrayController.addTabButton"].tap()
+        // Perform a search to check
+        XCUIApplication().buttons["TabTrayController.addTabButton"].tap()
+        
         app.textFields["url"].tap()
-
-        app.typeText("strange charm")
-        app.scrollViews.otherElements.buttons["Feeling Lucky search"].tap()
-        // Ensure that correct search is done
-        let url = app.textFields["url"].value as! String
-        XCTAssert(url.hasSuffix("&btnI"), "The URL should indicate that the search was performed using IFL")
+        app.typeText("window")
+        sleep(2)
```

Comment:
let's not use sleep().  instead, wait for the element you are asserting, like waitforNoExistence(app.scrollViews.otherElements.buttons["developer.mozilla.org search"])